core/base/explicitTriangulation/ExplicitTriangulation.cpp

```diff
@@ -309,6 +309,7 @@ int ExplicitTriangulation::preconditionEdgesInternal() {
 
 #ifdef TTK_ENABLE_MPI
     if(this->getDimensionality() == 2 || this->getDimensionality() == 3) {
+      this->preconditionTriangleEdges();
```

Comment:
is this call really needed?

if this call is indeed required in distributed mode, then, I would suggest to move it inside the call `preconditionDistributedEdges()`.

Reply:
moreover, if this call is indeed needed, it should be needed in the other triangulation types (implicit, etc.)
if this is the case, thanks for adding it to the `preconditionEdgesInternal` of the implicit triangulation.
(please remind me to discuss the periodic implicit triangulation and the implicit triangulations with preconditions)